reposerver/repository/repository.go

```diff
@@ -1244,18 +1244,35 @@ func helmTemplate(appPath string, repoRoot string, env *v1alpha1.Env, q *apiclie
 			}
 
 			if len(reposNotPermitted) > 0 {
-				return nil, status.Errorf(codes.PermissionDenied, "helm repos %s are not permitted in project '%s'", strings.Join(reposNotPermitted, ", "), q.ProjectName)
+				return nil, "", status.Errorf(codes.PermissionDenied, "helm repos %s are not permitted in project '%s'", strings.Join(reposNotPermitted, ", "), q.ProjectName)
 			}
 
-			return nil, err
+			return nil, "", err
 		}
 
-		out, err = h.Template(templateOpts)
+		out, command, err = h.Template(templateOpts)
 		if err != nil {
-			return nil, err
+			return nil, "", err
 		}
 	}
-	return kube.SplitYAML([]byte(out))
+	objs, err := kube.SplitYAML([]byte(out))
+
+	redactedCommand := redactPaths(command, gitRepoPaths, templateOpts.ExtraValues)
+
+	return objs, redactedCommand, err
+}
+
+func redactPaths(s string, paths io.TempPaths, extraValuesPath pathutil.ResolvedFilePath) string {
```

Comment:
Maybe add a godoc here. I am uncertain about what `extraValuesPath` is and why we are replacing it with `<temp file with override values>`

Reply:
Added some more explanation to the doc strings. Basically, the contents of `spec.source.helm.values` is dumped into a temporary, randomly-named file and passed to `helm template` via `--values <random path>`.

Since the random path isn't really helpful to the user, I'm just inserting a placeholder so they know what happened.

In practice, I don't expect many people to use `source.helm.values` in the source hydrator, so I don't think this will have a major UX impact.